### PR TITLE
prog builder enhancements

### PR DESCRIPTION
**Pull request by naazy**
prog builder enhancements https://github.com/OxfordAbstracts/application/issues/4666
https://github.com/OxfordAbstracts/application/issues/4676 delete/edit info bug

### CODE REVIEW

**Comment by naazy** [comment]
Thanks @nelsonic 🌟 🌮

File: src/MainUpdate.elm [hunk 1]
```diff
@@ -890,12 +890,26 @@ update msg model =
...
-                )
+                let
+                    highestSavedFileId =
+                        model.savedFiles
+                            |> List.map .id
+                            |> List.maximum
+                            |> Maybe.withDefault 0
+
+                    newfileToSaveWithId =
+                        FileToSave (highestSavedFileId + List.length model.filesToSave + 1)
```
Comment (roryc89) [inline src/MainUpdate.elm]: does the id need to be the highest plus length of filesToSave plus 1? Not just highest plus one?

File: src/MainUpdate.elm [hunk 2]
```diff
@@ -1180,6 +1212,28 @@ update msg model =
                 in
                     ( { model | filesToSave = newFilesToSave }, Cmd.none )
 
+            ChangeFileToSaveDescription fileToSaveId fileDescription ->
+                let
```
Comment (roryc89) [inline src/MainUpdate.elm]: let expression could be simplified to:
```
newFilesToSave =
  model.filesToSave 
    |> List.Extra.updateIf (.id >> (==) fileToSaveId) (\f -> { f | filedescription = fileDescription })
```

File: src/MainUpdate.elm [hunk 3]
```diff
@@ -1163,13 +1178,30 @@ update msg model =
                 in
                     ( { model | savedFiles = newSavedFiles }, Cmd.none )
 
+            ChangeSavedFileDescription savedFileId fileDescription ->
+                let
+                    fileToChange =
```
Comment (roryc89) [inline src/MainUpdate.elm]: same suggestion here as below

File: src/MainUpdate.elm (hunk 1 above)
Comment (naazy) [reply src/MainUpdate.elm]: Yeah that's correct. Because sometimes there will be multiple filesToSave

**Pull request merged**